doc/user_guide/data.rst

```diff
@@ -68,6 +68,21 @@ Similarly, we must also specify the data type when referencing data by URL:
         y='Miles_per_Gallon:Q'
     )
 
+You can use any URL visible to your frontend (i.e. JupyterLab, Jupyter Notebook, Colab, VSCode, Streamlit, etc.)
+
+All of those frontends have different mechanisms for accessing local files (if they have a mechanism at all), so there is no way to provide general advice for how to proceed. By way of example, consider a JupyterLab setup with seattle-weather.csv and a notebook in the top level folder. In this case, the full URL needs to be given.
+
+.. altair-plot::
+
+    import altair as alt
+    url = 'http://localhost:8888/files/seattle-weather.csv'
+
+    alt.Chart(url).mark_bar().encode(
+        x='month(date):O',
+        y='count():Q',
+        color='weather:N'
+    )
+
```

Comment:
Let's not use an `altair-plot` directive here, as it will attempt to render a chart and fail due to the data URL not being present. Instead you can make it a standard RST code block.

Reply:
Also, `localhost:8888` is not always the place where files are served. We should preface this with more info, e.g. "For example, if you are using a local JupyterLab instance that is hosted on `http://localhost:8888`, then your URL might look something like this for a data file located in the same directory where jupyterlab was started:"

All these required caveats are why I've not previously included something like this in the docs.

Reply:
One question though: is the full URL required? I seem to remember that you could do something like `files/data.json`, though it's been a while since I've used Altair on Jupyterlab.

Reply:
Indeed, the full URL is not required in JupyterLab. See https://github.com/altair-viz/altair/issues/2432#issuecomment-808419696